File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.py
import sys
input = sys.stdin.readline

N, K = map(int, input().split())
arr = [[0,0]] + [list(map(int, input().split())) for _ in range(N)]
dp = [[0] * (K+1) for _ in range(N+1)]

for i in range(1, N+1):
    for j in range(1, K+1):
        if j < arr[i][0]:
            dp[i][j] = dp[i-1][j]
        else:
            dp[i][j] = max(dp[i-1][j], dp[i-1][j-arr[i][0]]+ arr[i][1])
# [print(*dp[i]) for i in range(N+1)]
print(dp[N][K])

File: node.js/백준/Gold/12865. 평범한 배낭/평범한 배낭.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, K] = input[0].split(' ').map(Number);
const items = input.slice(1).map(e=>e.split(' ').map(Number));
const dp = Array.from({length: N+1}, () => Array(K+1).fill(0));

for (let i = 1; i<N+1; i++) {
    for (let j= 1 ; j <= K; j++) {
        if (j < items[i-1][0]) {
            dp[i][j] = dp[i-1][j];
        } else {
            dp[i][j] = Math.max(dp[i-1][j], dp[i-1][j-items[i-1][0]]+items[i-1][1]);
        }
    }
}
console.log(dp[N][K]);